internal/build/lifecycle_execution.go

```diff
@@ -706,6 +744,133 @@ func (l *LifecycleExecution) ExtendRun(ctx context.Context, buildCache Cache, ph
 	defer extend.Cleanup()
 	return extend.Run(ctx)
 }
+func (l *LifecycleExecution) ExtendBuildByDaemon(ctx context.Context) error {
+	extendtime := time.Now()
+	builderImageName := l.opts.BuilderImage
+	defaultFilterFunc := func(file string) bool { return true }
+	var extensions Extensions
+	time1 := time.Now()
+	extensions.SetExtensions(l.tmpDir, l.logger)
+	l.logger.Debugf("extensions.SetExtensions for build took %s", time.Since(time1))
+	time2 := time.Now()
+	dockerfiles, err := extensions.DockerFiles(DockerfileKindBuild, l.tmpDir, l.logger)
+	if err != nil {
+		return fmt.Errorf("getting %s.Dockerfiles: %w", DockerfileKindBuild, err)
+	}
+	l.logger.Debugf("extensions.DockerFiles for build took %s", time.Since(time2))
+	dockerapplytime := time.Now()
+	for _, dockerfile := range dockerfiles {
+		buildContext := archive.ReadDirAsTar(filepath.Dir(dockerfile.Path), "/", 0, 0, -1, true, false, defaultFilterFunc)
+		buildArguments := map[string]*string{}
+		if dockerfile.WithBase == "" {
+			buildArguments["base_image"] = &builderImageName
+		}
+		buildOptions := types.ImageBuildOptions{
+			Context:    buildContext,
+			Dockerfile: "Dockerfile",
+			Tags:       []string{"newbuilder-image"},
+			Remove:     true,
+			BuildArgs:  buildArguments,
+		}
+		response, err := l.docker.ImageBuild(ctx, buildContext, buildOptions)
+		if err != nil {
+			return err
+		}
+		defer response.Body.Close()
+		_, err = io.Copy(os.Stdout, response.Body)
+		if err != nil {
+			return err
+		}
+		l.logger.Debugf("build response for the extend: %v", response)
```

Comment:
Can we also log warnings if the extension leaves the user as root and/or if the final user is different from the original user? See here: https://github.com/buildpacks/lifecycle/blob/e06b69b5acb14e441659178b13de4a38e7a2e612/extender.go#L354 and here: https://github.com/buildpacks/lifecycle/blob/e06b69b5acb14e441659178b13de4a38e7a2e612/extender.go#L361

Reply:
Did we add the warnings anywhere?

Reply:
not yet, then we need to inspect the image as well

Reply:
I think that makes sense - we do this in the lifecycle here: https://github.com/buildpacks/lifecycle/blob/aeb177ca468cc814ab47fac0cb878497495b5819/extender.go#L329

Reply:
added!